### ISSUE TITLE
Re-implement support for line item messaging

### ISSUE THREAD

**Issue by chrisaligent** [aligent/orocommerce-fees-bundle#17]
During the process of refactoring this bundle for open-source, LineItem Fee messaging was removed.

This feature allowed Fees to display a message below every applicable LineItem. For example, if a Handling Charge is being injected into the Checkout, an Oro validation message can be displayed below it to advise the Customer why it was added, and potentially explain how much more they can spend in order to remove it from the Checkout.

This would require adding a new `FeeLineItemInterface` which returns a warning, eg:
```php
use Oro\Bundle\CheckoutBundle\Entity\Checkout;

interface FeeLineItemMessagingInterface
{
    /**
     * Return a string containing a warning message to be displayed
     * below the fee during the checkout
     *
     * @param Checkout $checkout
     * @return string|null
     */
    public function getWarning(Checkout $checkout): ?string;
}
```

An EventListener would also be needed on the `line_item.validate` event, this would load all the registered Fees and determine whether a warning needed to be applied.
For example:
```php
use Aligent\FeesBundle\Fee\FeeLineItemMessagingInterface;
use Aligent\FeesBundle\Fee\FeeRegistry;
use Oro\Bundle\CheckoutBundle\Entity\Checkout;
use Oro\Bundle\CheckoutBundle\Entity\CheckoutLineItem;
use Oro\Bundle\ShoppingListBundle\Entity\LineItem;
use Oro\Bundle\ShoppingListBundle\Event\LineItemValidateEvent;
use Oro\Bundle\WorkflowBundle\Entity\WorkflowItem;
use Traversable;

class OrderLineItemFeeMessageListener
{
    public const ALLOWED_STEPS = [
        'order_review' => true,
        'checkout' => true,
        'enter_billing_address' => true,
        'enter_shipping_address' => true,
        'enter_shipping_method' => true,
        'enter_payment' => true
    ];

    /**
     * @var FeeRegistry
     */
    protected $feeRegistry;

    /**
     * @param FeeRegistry $feeRegistry
     */
    public function __construct(
        FeeRegistry $feeRegistry
    ) {
        $this->feeRegistry = $feeRegistry;
    }

    /**
     * Called when validating a LineItem
     *
     * @param LineItemValidateEvent $event
     */
    public function onLineItemValidate(LineItemValidateEvent $event)
    {
        // The ShoppingList has no Context, exclude it
        $context = $event->getContext();
        if (!$this->isContextSupported($context)) {
            return;
        }

        /** @var LineItem[] $lineItems */
        $lineItems = $event->getLineItems();

        if (!$lineItems instanceof Traversable) {
            return;
        }

        /** @var Checkout $checkout */
        $checkout = $context->getEntity();

        // Build a lookup array of all fee Products and their associated SKU
        $fees = [];
        foreach ($this->feeRegistry->getFees() as $fee) {
            $sku = $fee->getSKU();
            if ($sku) {
                $fees[$sku] = $fee;
            }
        }

        foreach ($lineItems as $lineItem) {
            if (!$lineItem instanceof CheckoutLineItem) {
                continue;
            }

            $sku = $lineItem->getProductSku();
            if (isset($fees[$sku]) && $fees[$sku] instanceof FeeLineItemMessagingInterface) {
                // This Fee has a Warning Message which needs to be injected against the Line Item
                $warning = $fees[$sku]->getWarning($checkout);

                if ($warning) {
                    // Add the warning message to the Fee LineItem
                    $event->addWarningByUnit(
                        $lineItem->getProductSku(),
                        $lineItem->getProductUnitCode(),
                        $warning
                    );
                }
            }
        }
    }

    /**
     * @param mixed $context
     * @return bool
     */
    protected function isContextSupported($context): bool
    {
        return $context instanceof WorkflowItem
            && $context->getEntity() instanceof Checkout
            && !empty(self::ALLOWED_STEPS[$context->getCurrentStep()->getName()]);
    }
}
```

**Comment by chrisaligent** [comment]
Will be resolved by #19 
![2022-08-10_13-39](https://user-images.githubusercontent.com/45612883/184055973-8028dab8-5325-4b8f-a46c-3e3bffcfda03.png)
![2022-08-10_13-38](https://user-images.githubusercontent.com/45612883/184055976-26823220-17ea-4663-963e-7945984c268a.png)

**Comment by chrisaligent** [comment]
Released in `5.0.3`
